src/Install/Prerequisite/Composite.php

```diff
@@ -25,21 +27,14 @@ public function __construct(PrerequisiteInterface $first)
         }
     }
 
-    public function check()
+    public function problems(): Collection
     {
         return array_reduce(
             $this->prerequisites,
-            function ($previous, PrerequisiteInterface $prerequisite) {
-                return $prerequisite->check() && $previous;
+            function (Collection $errors, PrerequisiteInterface $condition) {
+                return $errors->concat($condition->problems());
             },
-            true
+            collect()
```

Comment:
What is the benefit here of using the function `collect()` over direct class instantiation? The class is imported in this code already anyway?

Reply:
None, just laziness. Do you have a preference?

Reply:
My preference is to use class instantiation over function. But then again, maybe we need to have a stricter guideline for this.